controller/api/destination/watcher/endpoints_watcher.go

```diff
@@ -1081,3 +1085,22 @@ func isValidSlice(es *discovery.EndpointSlice) bool {
 
 	return true
 }
+
+// SetPodOpaquePortAnnotation ensures that if there is no opaque port
+// annotation on the pod, then it inherits the annotation from the namespace.
+// If there is also no annotation on the namespace, then it remains unset.
+func SetPodOpaquePortAnnotation(k8sAPI *k8s.API, pod *corev1.Pod, ns string) error {
+	annotation := pod.Annotations[consts.ProxyOpaquePortsAnnotation]
```

Comment:
Do we want to condition on the absence of this annotation instead of it being the empty string?  e.g. if the pod sets this annotation to the empty string, shouldn't that take priority over the namespace annotation and essentially unset all the opaque ports?

Reply:
Yep good point.